megamek/src/megamek/client/ui/swing/CommonSettingsDialog.java

```diff
@@ -1609,14 +1617,48 @@ private JPanel getOverlaysPanel() {
         return createSettingsPanel(comps);
     }
 
+    private BufferedImage boardImage;
+    private JLabel boardImageLabel;
```

Comment:
If these are only referenced from `getMiniMapPanel()`, could these variables be local scope to the method instead of class variables?  If not, do they need to be initialized?

Reply:
No, they cant, they have to be local to the dialog, otherwise the image wont ever be updated, as the reference to the object in the UI will be lost once you redesignate the local variable (the local  variable would have to be declared final, which makes it impossible to do what I am doing)

Reply:
Got it.  I just didn't see it being referenced anywhere, but this makes sense.